specs/core/0_beacon-chain.md

```diff
@@ -1170,6 +1171,17 @@ def integer_squareroot(n: int) -> int:
     return x
 ```
 
+#### `entry_exit_effect_slot`
+
+```python
+def entry_exit_effect_slot(n: int) -> int:
+    """
+    An entry or exit triggered in the slot given by the input takes effect at
+    the slot given by the output.
+    """
+    return (n - n % EPOCH_LENGTH) + EPOCH_LENGTH + ENTRY_EXIT_DELAY
```

Comment:
Does this work when `n % EPOCH_LENGTH == 0`? In this case `entry_exit_effect_slot(n)` returns `n + EPOCH_LENGTH + ENTRY_EXIT_DELAY` when it should return `n + ENTRY_EXIT_DELAY`.

Reply:
Is that what it should return? If it gets triggered during block `EPOCH_LENGTH * k`, then that means that the epoch transition for epoch `k` has already been processes, and so we are in epoch `k+1` now, so returning `(EPOCH_LENGTH * k) + EPOCH + ENTRY_EXIT_DELAY` is the _correct_ thing to do.

Reply:
agreed